### PR TITLE
Swagger-codegen library

### PR DESCRIPTION
**Pull request by meliaj**
Generated library from our swagger api file for v1.2.

Added in logic to handle our event types and a test for this.

Major version bump to 1.0.0

### CODE REVIEW

**Comment by meliaj** [comment]
Jumped straight to 2.0

File: README.md [hunk 1]
```diff
@@ -34,133 +32,74 @@ Composer support has been added,  Place the following lines in your composer.jso
 }
 ```
 
+Then run `composer install`
+
 **Note that it is necessary to include the vcs link to be sure that you are pulling from this repo.**
 
 The package manager for composer currently does **not** point to the correct repo for [clever/clever-php](https://packagist.org/packages/clever/clever-php).
```
Comment (kvigen) [inline README.md]: can we fix this?

File: README.md [hunk 2]
```diff
@@ -34,133 +32,74 @@ Composer support has been added,  Place the following lines in your composer.jso
...
-} while($last_school);
-print_r($schools); # Print all the schools retrieved.
-```
+1. Git clone swagger-codegen (https://github.com/swagger-api/swagger-codegen)
 
-To cursor to a previous page, use the first ID of each set as the `ending_before` value.
+2. Git clone Clever's swagger-api repo (https://github.com/Clever/swagger-api)
 
-Unfortunately, paginating via relative links is not yet supported by clever-php.
+3. Are you removing some old models? Consider removing existing files from `docs/` and `lib/` and `/test` before regenerating files
```
Comment (kvigen) [inline README.md]: should we just delete these files as part of the generation?

File: README.md (hunk 1 above)
Comment (meliaj) [reply README.md]: 🤷‍♀️  It's been like that forever. May as part of a separate PR

File: README.md (hunk 2 above)
Comment (meliaj) [reply README.md]: Done

**Comment by meliaj** [comment]
ok so i’d:
- lost my events test
- didn’t commit overridden changes to ObjectSerializer
- had to fix the way we set the token for tests to pass

Now fixed all of that

**Pull request merged**